public/app/plugins/datasource/dashboard/datasource.ts

```diff
@@ -104,7 +105,19 @@ export class DashboardDatasource extends DataSourceApi<DashboardQuery> {
         },
       }));
     } else {
-      return [...data.series, ...annotations];
+      const series = data.series.map((s) => {
```

Comment:
can we just cloneDeep?

Reply:
My idea was to make sure the field state gets cloned properly and shallow clone everything else, I'd reckon `cloneDeep` might potentially cause some performance issues, thus going for this approach.